kinto/plugins/history/views.py

```diff
@@ -43,3 +45,88 @@ def _extract_filters(self, queryparams=None):
+        filters = super(Revert, self)._extract_filters(request.params)
+        filters = [s for s in filters
+                                 if s.field != 'since']
+        self.storage = request.registry.storage
+        self.history, _ = self.storage.get_all(parent_id='/buckets/%s' % self.bucket_id,
+                                               collection_id='history', filters=filters,
+                                               sorting=[Sort('last_modified', -1)])
+
+    def batch_revert(self, revert_list):
+        batch = {'path': '/batch', 'method': 'POST', 'body': {'requests': revert_list}}
+        subrequest = build_request(self.request, batch)
+        try:
+            resp, subrequest = self.request.follow_subrequest(subrequest,
+                                                         use_tweens=False)
+        except httpexceptions.HTTPException as e:
+            if e.content_type == 'application/json':
+                resp = e
+            else:
+                resp = errors.http_error(e)
+        return build_response(resp, subrequest)
+
+    def revert_changes(self):
+        revert_list = self.build_revert_list()
+        if len(revert_list) == 0:
+            raise_invalid(self.request, 'params', 'no history', 'there are no history records to revert after since timestamp')
+        return self.batch_revert(revert_list)
+
+    def create_revert_operation(self, last, first, passed_since):
+        path = last['uri']
+        last_is_create = last['action'] == ACTIONS.CREATE.value
+        if last_is_create or (not passed_since and first['action'] == ACTIONS.CREATE.value):
+            return {'method': 'DELETE', 'path': path }
+        else:
+            body = first['target']
+            return {'method': 'PUT', 'path': path, 'body':  body }
+
+    def insert_rec_to_revert_lists(self, rec, revert_lists, passed_since):
+        rec_id = rec['target']['data']['id']
+        if passed_since:
+            if rec_id not in revert_lists['not_paired']:
+                return
+            last_change = revert_lists['not_paired'].pop(rec_id)
+            revert_lists['paired'].append(self.create_revert_operation(last_change, rec, passed_since))
+        else:
+            if rec_id in revert_lists['not_paired']:
+                if rec['action'] == ACTIONS.CREATE.value:
+                    last_change = revert_lists['not_paired'].pop(rec_id)
+                    revert_lists['paired'].append(self.create_revert_operation(last_change, rec, passed_since))
+            elif rec['action'] == ACTIONS.CREATE.value:
+                revert_lists['paired'].append(self.create_revert_operation(rec, None, passed_since))
+            else:
+                revert_lists['not_paired'][rec_id] = rec
+
+    def build_revert_list(self):
+        revert_lists = {'not_paired': {}, 'paired': []}
+        passed_since = False
+        for rec in self.history:
+            if not passed_since and rec['last_modified'] < self.since:
```

Comment:
Can you explain this `passed_since` boolean in comments? I see that you look-up some entries in the history, that you reconciliate later, but this is not easy to grasp